odbc/src/api/errors.rs

```diff
@@ -1,49 +1,50 @@
-const VENDOR_IDENTIFIER: &str = "MongoDB";
+use constants::{HY024, HYC00, VENDOR_IDENTIFIER, _01S02};
+use thiserror::Error;
 
-// SQL states
-pub const HYC00: &str = "HYC00";
-pub const HY024: &str = "HY024";
-pub const _01S02: &str = "01S02";
-
-#[derive(Debug)]
+#[derive(Debug, Error)]
 pub enum ODBCError {
+    #[error("[{}][API] The feature {0} is not implemented", VENDOR_IDENTIFIER)]
     Unimplemented(&'static str),
+    #[error("[{}][API] Invalid Uri {0}", VENDOR_IDENTIFIER)]
+    InvalidUriFormat(String),
+    #[error("[{}][API] Invalid handle type, expected {0}", VENDOR_IDENTIFIER)]
+    InvalidHandleType(&'static str),
+    #[error("[{}][API] Invalid value for attribute {0}", VENDOR_IDENTIFIER)]
     InvalidAttrValue(&'static str),
+    #[error(
+        "[{}][API] Invalid value for attribute {0}, changed to {1}",
+        VENDOR_IDENTIFIER
+    )]
     OptionValueChanged(&'static str, &'static str),
+    #[error("[{}][Core] {0}", VENDOR_IDENTIFIER)]
+    CoreError(#[from] mongo_odbc_core::Error),
```

Comment:
```suggestion
    Core(#[from] mongo_odbc_core::Error),
```
[nit] Should we omit the suffix "Error" from this variant name since "Error" is implied by the enum name itself?

Reply:
Good idea!